pandas/core/frame.py

```diff
@@ -6004,7 +6004,12 @@ def dropna(
             else:
                 raise TypeError("must specify how or thresh")
 
-        result = self.loc(axis=axis)[mask]
+        if all(mask):
```

Comment:
this is too specific a place for this, it needs to live in the the indexing logic itself. you will have to debug thru the calls and see if you can find a good place w/o breaking anything else (as this likely IS a breaking change).